### ISSUE TITLE
Replace Nginx HTTP Auth with Google OAuth

### ISSUE THREAD

**Issue by rahul286** [EasyEngine/easyengine#992]
**Please refer**
https://github.com/cloudflare/nginx-google-oauth
https://github.com/bitly/oauth2_proxy

This is mainly for agencies and product companies which use Google Apps/G Suite. 

Say on a client/internal demo/dev/staging server, rtCamp put `rtcamp.com` domain in ee.conf for some key then all or selective sites on that server should be only accessible for rtcamp.com employee who can log in from a google account with rtcamp.com domain.

Of course, multiple domains and individual Gmail ids can be supported.

Also, there should be a way to fallback to HTTP Auth with username/password so if there is a non-google user, they wont be denied service!

**Comment by alexlii1971** [comment]
"Also, there should be a way to fallback to HTTP Auth with username/password so if there is a non-google user, they wont be denied service!"

I am just quite doubt whether it is good for the users in some countries, like China, Japan...

we have servers both in China and Japan. In China, we are blocked by Chinese government firewall to access any global resource like google, Facebook, twitter, yahoo, Skype, wordpress.com....and I need user VPN to access wordpress.org everyday.

In Japan, the site also be caused delay if there is some google JS resource in site, and we have put any Js resource in local server.

so, please take these into consideration.

thanks.

**Comment by mbtamuli** [comment]
@alexlii1971 If I understand correctly, you want your access through VPN be uninterrupted? Well, you can surely exclude/whitelist the VPN IP address(es) for HTTP Auth as shown here - https://easyengine.io/docs/commands/secure/

**Comment by s-a-s-k-i-a** [comment]
Please take european GDPR into consideration when implementing Google Services in any way.
Here in EU it has become a legal issue even implementing Google fonts from google cdn in our websites. We must selfhost our google fonts to be legally safe...

**Comment by rahul286** [comment]
@s-a-s-k-i-a of course, Google Auth (SSO) will be an EE package and something server admin will need to install manually. It will have some configuration step also.

Cloudflare recently launched something similar https://www.cloudflare.com/products/cloudflare-access/
